include/oneapi/dpl/pstl/algorithm_impl.h

```diff
@@ -1691,19 +1691,25 @@ __pattern_reverse(_ExecutionPolicy&&, _BidirectionalIterator __first, _Bidirecti
                   _IsVector _is_vector,
                   /*is_parallel=*/::std::false_type) noexcept
 {
-    __internal::__brick_reverse(__first, __last, _is_vector);
+    if (__first != __last)
+    {
+        __internal::__brick_reverse(__first, __last, _is_vector);
+    }
 }
 
 template <class _ExecutionPolicy, class _RandomAccessIterator, class _IsVector>
 oneapi::dpl::__internal::__enable_if_host_execution_policy<_ExecutionPolicy>
 __pattern_reverse(_ExecutionPolicy&& __exec, _RandomAccessIterator __first, _RandomAccessIterator __last,
                   _IsVector __is_vector, /*is_parallel=*/::std::true_type)
 {
-    __par_backend::__parallel_for(
-        ::std::forward<_ExecutionPolicy>(__exec), __first, __first + (__last - __first) / 2,
-        [__is_vector, __first, __last](_RandomAccessIterator __inner_first, _RandomAccessIterator __inner_last) {
-            __internal::__brick_reverse(__inner_first, __inner_last, __last - (__inner_first - __first), __is_vector);
-        });
+    if (__first != __last)
```

Comment:
I would suggest return on trivial case like in other patterns, f.e. https://github.com/oneapi-src/oneDPL/pull/1442/files#diff-39feab6040f4974b495666d46a63d64a1fcbe9fcb836d5bad9d645d1a6129184L1996

In that case we dont need to have indentations for many lines where a parallel backend pattern is called.

Reply:
Fixed